src/commonMain/kotlin/exchange.dydx.abacus/calculator/TradeInputCalculator.kt

```diff
@@ -1718,7 +1688,20 @@ internal class TradeInputCalculator(
                 "openPositions.$marketId",
             ),
         )
-        return maxLeverageFromPosition(position, market)
+        val equity = parser.asDouble(parser.value(subaccount, "equity.current"))
+        val freeCollateral = parser.asDouble(parser.value(subaccount, "freeCollateral.current"))
+
+        val initialMarginFraction =
+            parser.asDouble(parser.value(market, "configs.effectiveInitialMarginFraction"))
+                ?: return null
+        val maxMarketLeverage = Numeric.double.ONE / initialMarginFraction
```

Comment:
doubt it'd happen, but for safety's sake
```suggestion
        val maxMarketLeverage = if (initialMarginFraction <= 0.0) {
    return null
} else {
    Numeric.double.ONE / initialMarginFraction
}
```